pkg/certificateauthority/caconfigmap.go

```diff
@@ -56,9 +57,21 @@ func MountCaConfigMap(deployment *appsv1.Deployment) {
 // getCaOnDisk returns the contents of the Certificate Authority bundle on disk as a byte
 // array or returns the error encountered when attempting to do so.
 func getCaOnDisk() ([]byte, error) {
-	b, err := ioutil.ReadFile(trustedCaMountPath + caBundlePath)
+	return getFileOnDisk(trustedCaMountPath + caBundlePath)
+}
+
+// getCaOnDisk returns the contents of the Certificate Authority bundle on disk as a byte
+// array or returns the error encountered when attempting to do so.
+func getFileOnDisk(path string) ([]byte, error) {
+	b, err := ioutil.ReadFile(path)
 	if err != nil {
 		return []byte{}, err
 	}
 	return b, nil
 }
+
+// isRunningOnPod returns true if marketplace is being ran on a pod.
+func isRunningOnPod() bool {
+	_, err := getFileOnDisk("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
Question: Is Checking the existence of `/var/run/secrets/kubernetes.io/serviceaccount/namespace` a convention the community uses to check whether the process is running within a pod?

Reply:
I had gotten the suggestion from @joelanford, I have not seen a community convention.